votor/src/event_handler/stats.rs

```diff
@@ -0,0 +1,312 @@
+        }
+        datapoint_info!(
+            "event_handler_timing",
+            (
+                "receive_event_time_us",
+                self.receive_event_time_us as i64,
+                i64
+            ),
+            (
+                "send_votes_batch_time_us",
+                self.send_votes_batch_time_us as i64,
+                i64
+            ),
+        );
+        for (vote_type, count) in &self.sent_votes {
+            datapoint_info!(
+                "event_handler_sent_vote_count",
+                ("vote", format!("{:?}", vote_type), String),
+                ("count", *count as i64, i64)
+            );
+        }
+        // Only report if the slot is lower than root_slot
+        let split_off_map = self.slot_tracking_map.split_off(&self.root_slot);
+        for (slot, tracking) in &self.slot_tracking_map {
+            let start = tracking.start;
+            datapoint_info!(
+                "event_handler_slot_tracking",
+                ("slot", *slot as i64, i64),
+                (
+                    "first_shred",
+                    tracking.first_shred.map(|t| {
+                        t.saturating_duration_since(start)
+                            .as_micros()
+                            .min(i64::MAX as u128) as i64
+                    }),
+                    Option<i64>
+                ),
+                (
+                    "parent_ready",
+                    tracking.parent_ready.map(|t| {
+                        t.saturating_duration_since(start)
+                            .as_micros()
+                            .min(i64::MAX as u128) as i64
+                    }),
+                    Option<i64>
+                ),
+                (
+                    "vote_notarize",
+                    tracking.vote_notarize.map(|t| {
+                        t.saturating_duration_since(start)
+                            .as_micros()
+                            .min(i64::MAX as u128) as i64
+                    }),
+                    Option<i64>
+                ),
+                (
+                    "vote_skip",
+                    tracking.vote_skip.map(|t| {
+                        t.saturating_duration_since(start)
+                            .as_micros()
+                            .min(i64::MAX as u128) as i64
+                    }),
+                    Option<i64>
+                ),
+                (
+                    "finalized",
+                    tracking.finalized.map(|t| {
+                        t.0.saturating_duration_since(start)
+                            .as_micros()
+                            .min(i64::MAX as u128) as i64
+                    }),
+                    Option<i64>
+                ),
+                ("is_fast_finalization", tracking.finalized.map(|t| t.1), Option<bool>)
+            );
+        }
+        self.last_report_time = now;
+        let root_slot = self.root_slot;
+        *self = EventHandlerStats::new();
```

Comment:
Hmm, maybe we should have a new which takes root_slot and slot_tracking_map.

Reply:
adding `fn reset` that takes `root_slot` and `slot_tracking_map`